File: Server/PHP/mobo.php
<?php
require_once("connection.php");
require_once("common.php");

$db = new dbObj();
$connection = $db->getConnection();
$request_method = $_SERVER["REQUEST_METHOD"];

switch($request_method){
    case 'GET':         
        if(!empty($_GET["id"])){
            $id = intval($_GET["id"]);
            $data = get_mobo_by_id($id);
            echo json_encode($data);
        }else
            $data = get_all_mobos();
            echo json_encode($data);
    break;
    case 'POST':
        $content = file_get_contents('php://input');
        $data = json_decode($content, true);
        $user = checkLoggedIn($data["username"], $data["password"]);
        $admin = checkAdmIn($data["username"], $data["password"]);
        if(!$user or !$admin){
            header('HTTP/1.0 401 Unauthorized ');
            break;
        }
        insert_mobo($data["manufacturer"], $data["model"], $data["ram_type"], $data["ram_sockets"], $data["price"]);
    break;
    
    case 'PUT': 
        $content = file_get_contents('php://input');
        $data = json_decode($content, true);
        $user = checkLoggedIn($data["username"], $data["password"]);
        $admin = checkAdmIn($data["username"], $data["password"]);
        if(!$user or !$admin){
            header('HTTP/1.0 401 Unauthorized ');
            break;
        }
        update_mobo($data["id"], $data["manufacturer"], $data["model"], $data["ram_type"], $data["ram_sockets"], $data["price"]);
    break;
    
    case 'DELETE': 
        $content = file_get_contents('php://input');
        $data = json_decode($content, true); // asszociatív tömb
        $user = checkLoggedIn($data["username"], $data["password"]);
        $admin = checkAdmIn($data["username"], $data["password"]);
        if(!$user or !$admin) {
            header('HTTP/1.0 401 Unauthorized ');
            break;
        }
        delete_mobo($data["id"]);
    break;
    
    default:
        header('HTTP/1.1 405 Method Not Allowed');
        header('Allow: GET, POST, PUT, DELETE');
    break;
}

function get_all_mobos(){
    global $connection;
    
    // Perform query
    $result = $connection -> query("SELECT id, manufacturer, model, ram_type, ram_sockets, price FROM mobo");
    
    return $result->fetch_all(MYSQLI_ASSOC);
}

function get_mobo_by_id($id=0){
    global $connection;
    
    // Perform query
    $result = $connection -> query("SELECT id, manufacturer, model, ram_type, ram_sockets, price FROM mobo WHERE id = '$id'");
    
    return $result->fetch_all(MYSQLI_ASSOC);
}

function insert_mobo($manufacturer, $model, $ram_type, $ram_sockets, $price){
    global $connection;
    
    // Perform query
    $result = $connection -> query("INSERT INTO mobo SET id = default, manufacturer = '$manufacturer', model = '$model', ram_type = '$ram_type', ram_sockets = '$ram_sockets', price = '$price'");
}

function update_mobo($id, $manufacturer, $model, $ram_type, $ram_sockets, $price){
    global $connection;

    // Perform query
    $connection -> query("UPDATE mobo SET manufacturer = '$manufacturer', model = '$model', ram_type = '$ram_type', ram_sockets = '$ram_sockets', price = '$price' WHERE id = '$id'");
}

function delete_mobo($id){
    global $connection;

    // Perform query
    $connection -> query("DELETE FROM mobo WHERE id = '$id'");
}

?>

File: Client/O48WRX_RESTFULCLIENT/O48WRX_RESTFULCLIENT/Classes/Mobo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace O48WRX_RESTFULCLIENT.Classes
{
    class Mobo
    {
        private int id;

        public int Id
        {
            get { return id; }
            set { id = value; }
        }

        private string manufacturer;

        public string Manufacturer
        {
            get { return manufacturer; }
            set { manufacturer = value; }
        }

        private string model;

        public string Model
        {
            get { return model; }
            set { model = value; }
        }


        private string ram_type;

        public string Ram_type
        {
            get { return ram_type; }
            set { ram_type = value; }
        }

        private int ram_sockets;

        public int Ram_sockets
        {
            get { return ram_sockets; }
            set { ram_sockets = value; }
        }

        private int price;

        public int Price
        {
            get { return price; }
            set { price = value; }
        }

        public Mobo()
        {

        }

        public Mobo(int id, string manufacturer, string model, string ram_type, int ram_sockets, int price)
        {
            this.Id = id;
            this.Manufacturer = manufacturer;
            this.Model = model;
            this.Ram_type = ram_type;
            this.Ram_sockets = ram_sockets;
            this.Price = price;
        }
    }
}
